### PR TITLE
[Instrumentation.MySqlData] Nullable

### PR DESCRIPTION
**Pull request by Kielek**
Towards #894.

## Changes

[Instrumentation.MySqlData] Nullable.
Part of shared files also updated.

For significant contributions please make sure you have completed the following items:

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~
* ~~[ ] Changes in public API reviewed~~

### CODE REVIEW

**Comment by moonheart** [comment]
This LGTM 👍

File: src/OpenTelemetry.Instrumentation.MySqlData/TracerProviderBuilderExtensions.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public static class TracerProviderBuilderExtensions
     /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
     public static TracerProviderBuilder AddMySqlDataInstrumentation(
         this TracerProviderBuilder builder,
-        Action<MySqlDataInstrumentationOptions> configure)
+        Action<MySqlDataInstrumentationOptions>? configure)
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.MySqlData/TracerProviderBuilderExtensions.cs:44]: We should probably not put `?` to stay consistent with other instrumentation libraries.

Look at this for reference: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/TracerProviderBuilderExtensions.cs#L59-L71

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.MySqlData/TracerProviderBuilderExtensions.cs:44]: I have discussion with @alanwest some time ago. The consensus is to make it nullable non-optional.

See https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/913

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.MySqlData/TracerProviderBuilderExtensions.cs:44]: Ah okay. Thanks for sharing that.

**Pull request merged**